Fixes issue #18: timex 3.7 changes meaning of `GMT` and `Etc/GMT`

suranyami opened the pull request:
Azure is expecting the `x-ms-date` header to have the trailing timezone code as `GMT`.

`timex` 3.7 requires the timezone code `Etc/GMT` if we want to keep the polarity of the offset the same.

To fix this, have changed the timezone to be `Etc/GMT` and `String.replace` it to `GMT`.

Also, added a test for the above.

suranyami commented:
Also, @jakobht Happy to be a maintainer on this project.

suranyami commented:
Hi. Silly me… the tests failed because of different function signatures in `DateTime` introduced in Elixir 1.11.

I've updated the versions in `.travis.yml` and `mix.exs` but not sure if it needs to be updated anywhere else for the Github CI actions to complete. Are you even using Travis? Or is it Github CI/Actions? @Zinoex

jakobht commented on .travis.yml:
```diff
@@ -2,8 +2,8 @@ sudo: false
 
-elixir: "1.9"
-opt_release: "22.0"
+elixir: "1.11"
```
First of all - thanks a lot for the PR!

The workflow is failing because you have bumped the elixir version - do we use something from 1.11? If not I think we should allow running on 1.9 as well (same for opt 22).

jakobht replied:
@suranyami

jakobht replied:
Ah sorry - this comment applies to the mix.exs file 
As far as I can see the workflow file used is this: 
https://github.com/jakobht/azurex/blob/0c5980f5b555dfec09fb35fffed0c4bfde66cb76/.github/workflows/elixir.yml
So if we really want to bump the version we should change that one - but again, I don't really see a reason to do that. What do you think @Zinoex ?

tofran commented:
@jakobht thank you for the project and David, thank you for the PR

What is the oldest elixir version that you want to support?

tofran commented:
In the meantime I have created a PR to the PR: https://github.com/workplacearcade/azurex/pull/1 : Removes `timex`

@jakobht  Whats your opinion?

suranyami commented:
* Have removed need for `timex` (thanks @tofran !)
* Reverted minimum version of elixir needed to 1.9 for better compatibility
* Updated the `SharedKeyTest` to use 1.9-compatible `DateTime` fixture

@jakobht

jakobht commented:
I made a new PR, see #21, as I'm not able to commit to this branch - see [Allowing changes to a pull request branch created from a fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
Thanks a lot for the PR and all the contributions, I'll merge the other PR this afternoon
